message/router.go

```diff
@@ -219,6 +275,41 @@ func (r *Router) Run() (err error) {
 		}
 	}
 
+	r.logger.Debug("Applying decorators", nil)
+	for name, handler := range r.handlers {
+		pub := handler.publisher
+		for _, decorator := range r.publisherDecorators {
```

Comment:
decorators for pub and sub should be probably applied in reversed order (like middlewares), because decorators added at the beginning should be executed at first

Reply:
9bf86d0855a0af420e5271cb9079119fe7f16e66 done

Reply:
with unit tests, too